### ISSUE TITLE
Need clarity on concept of key type, including info on using Hashing to create the key type

### ISSUE THREAD

**Issue by nicolehaugen** [dotnet/machinelearning#4041]
When learning about machine learning in general, I didn't see reference to a 'key type' outside of ML.NET - as a result, this concept was difficult for me to understand from the current description provided in the docs.

It would be better to provide an example that clarifies how and why a key type is needed.  The example that someone shared with me and that I found to be helpful, is this:
* When you use MapValueToKey, it "learns" the keys based on the training set.
* Using the U.S. States as an example, it "learns" that ND maps to 1, MN to 2, SD to 3, etc.
* And, when you see a new value in the test set that isn't in the training set, it will map it to an "unknown" key (e.g. 0)

Another way to make a KeyType is to use Hashing.  This should also be mentioned and explained in the docs alongside the description of KeyTypes.

**Comment by MichaelSimons** [comment]
Issue moved to [dotnet/docs #13590](https://github.com/dotnet/docs/issues/13590) via [**ZenHub**](https://www.zenhub.com/)

**Comment by nicolehaugen** [comment]
Closing since this was moved to docs: https://github.com/dotnet/docs/issues/13590.
